src/timely-util/src/reclock.rs

```diff
@@ -1007,4 +1017,115 @@ mod test {
         let b = ChainBatch::from_iter([('a', 0, -1), ('a', 1, 1)]);
         assert_eq!(a.merge_with(b), ChainBatch::from_iter([('a', 1, 1)]));
     }
+
+    #[mz_ore::test]
+    fn test_binding_consolidation() {
+        use std::sync::atomic::Ordering;
+
+        #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+        struct Time(u64);
+
+        // A counter of the number of active Time instances
+        static INSTANCES: AtomicUsize = AtomicUsize::new(0);
+
+        impl Time {
+            fn new(time: u64) -> Self {
+                INSTANCES.store(INSTANCES.load(Ordering::Relaxed) + 1, Ordering::Relaxed);
```

Comment:
Looks like the harness only runs a single worker, so this doesn't matter currently, but I think you want `INSTANCES.fetch_add(1, Ordering::Relaxed)` instead. At the very least it's more readable.